flows/vision-zero/cr3_ocr_narrative_extract_diagram/cr3_ocr_narrative_extract_diagram.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python3
+
+import os
+import json
+from subprocess import Popen, PIPE
+
+import prefect
+from prefect import Flow
+from prefect.backend import get_key_value
+from prefect.schedules.clocks import CronClock
+from prefect.run_configs import UniversalRun
+from prefect.tasks.shell import ShellTask
+
+kv_store = get_key_value("Vision Zero")
+kv_dictionary = json.loads(kv_store)
+
+HASURA_ENDPOINT = kv_dictionary["PRODUCTION_GRAPHQL_ENDPOINT"]
+HASURA_ADMIN_KEY = kv_dictionary["PRODUCTION_GRAPHQL_ENDPOINT_ACCESS_KEY"]
+AWS_DEFAULT_REGION = kv_dictionary["AWS_DEFAULT_REGION"]
+AWS_ACCESS_KEY_ID = kv_dictionary["AWS_ACCESS_KEY_ID"]
+AWS_SECRET_ACCESS_KEY = kv_dictionary["AWS_SECRET_ACCESS_KEY"]
+
+OCR_CR3_SOURCE_BUCKET = kv_dictionary["OCR_CR3_SOURCE_BUCKET"]
+OCR_CR3_SOURCE_PATH = kv_dictionary["OCR_CR3_SOURCE_PATH"]
+OCR_DIAGRAM_TARGET_BUCKET = kv_dictionary["OCR_DIAGRAM_TARGET_BUCKET"]
+OCR_DIAGRAM_TARGET_PATH = kv_dictionary["OCR_DIAGRAM_TARGET_PATH"]
+OCR_BATCH_SIZE = kv_dictionary["OCR_BATCH_SIZE"]
+OCR_SINGLE_CRASH = kv_dictionary["OCR_SINGLE_CRASH"]
+
+env = {
+    "HASURA_ENDPOINT": HASURA_ENDPOINT,
+    "HASURA_ADMIN_KEY": HASURA_ADMIN_KEY,
+    "AWS_DEFAULT_REGION": AWS_DEFAULT_REGION,
+    "AWS_ACCESS_KEY_ID": AWS_ACCESS_KEY_ID,
+    "AWS_SECRET_ACCESS_KEY": AWS_SECRET_ACCESS_KEY,
+}
+
+task = ShellTask(return_all=True, log_stderr=True, stream_output=True, env=env)
+
+with Flow(
+    "CR3 Narrative OCR and Diagram Extraction",
+    # schedule=Schedule(clocks=[CronClock("* * * * *")]),
+    run_config=UniversalRun(labels=["vision-zero", "atd-data03"]),
+) as flow:
+    logger = prefect.context.get("logger")
+
+    # see https://github.com/cityofaustin/atd-airflow/blob/master/dags/python_scripts/cr3_extract_diagram_ocr_narrative.py#L15-L25
+
+    op_mode = (
+        f"--crash-id {OCR_SINGLE_CRASH} "
+        if OCR_SINGLE_CRASH
+        else f"--batch {OCR_BATCH_SIZE} "
+    )
+    command = (
+        f"/usr/bin/python3 \
+        /root/cr3_ocr_narrative_extract_diagram/atd-airflow/dags/python_scripts/cr3_extract_diagram_ocr_narrative.py "
+        + op_mode
+        + f"-vd --update-narrative --update-timestamp \
+        --cr3-source {OCR_CR3_SOURCE_BUCKET} {OCR_CR3_SOURCE_PATH} \
+        --save-diagram-s3 {OCR_DIAGRAM_TARGET_BUCKET} {OCR_DIAGRAM_TARGET_PATH}"
+    )
+
+    stream = task(command=command)
+
+    logger.info(stream)
+
+#flow.run()
+flow.register(project_name="vision-zero")
```

Comment:
After registering a flow like this, does one then need to go into the UI to trigger the scheduled runs? 

Also, do you think adding an "idempotency_key" would be a good idea? https://docs.prefect.io/orchestration/concepts/flows.html#core-client 
Or would the `register` command be commented out and `flow.run()` be used after PR approval?

Reply:
Re scheduling:

Great catch! I do envision that the scheduling being done via the Prefect UI. The execution of the flow to register gets it queued up to be scheduled, and then the scheduling is contained in the UI, not dictated by code that may or may not be up to date. Seeking a single source of truth.

Reply:
Re the `indempotency_key` argument to the registration function:

TIL: indempotent - maths not changed in value following multiplication by itself. 

I think the jury is still out on if this will be needed. When I put this PR together, I was imagining the programmer being in charge of when things get registered and the agent getting rebuilt etc. As it was, I would just want to count on the developer to "do the right thing", but it was you and my discussion from the other day that prompted me to reevaluate that. After that conversation and with an issue put on deck for me by product, I am going to seek automation via GitHub actions for deployment related tasks. It'll be this work which will inform the need of the `indempotency_key`.